components/service-operator/controllers/image_repository_cloudformation_test.go

```diff
@@ -0,0 +1,159 @@
+				Name:      secretName,
+			}
+			principal = access.Principal{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: access.GroupVersion.Group,
+					Kind:       "Principal",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: namespace,
+					Name:      principalName,
+					Labels: map[string]string{
+						cloudformation.AccessGroupLabel: "test.access.group",
+					},
+				},
+			}
+			repository = storage.ImageRepository{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "ImageRepository",
+					APIVersion: storage.GroupVersion.Group,
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:         name,
+					GenerateName: "",
+					Namespace:    namespace,
+					Labels: map[string]string{
+						cloudformation.AccessGroupLabel: "test.access.group",
+					},
+				},
+				Spec: storage.ImageRepositorySpec{
+					Secret: secretName,
+				},
+			}
+			secret core.Secret
+		)
+
+		By("creating a prequisite Principal resource with kubernetes api", func() {
+			Expect(client.Create(ctx, &principal)).To(Succeed())
+		})
+
+		By("creating an ImageRepository resource with kubernetes api", func() {
+			Expect(client.Create(ctx, &repository)).To(Succeed())
+		})
+
+		By("displaying a READY resource status after initial creation", func() {
+			Eventually(func() object.State {
+				_ = client.Get(ctx, resourceNamespacedName, &repository)
+				return repository.GetState()
+			}, timeout).Should(Equal(object.ReadyState))
+		})
+
+		By("displaying an AWS CREATE_COMPLETE resource status after initial creation", func() {
+			Eventually(func() string {
+				_ = client.Get(ctx, resourceNamespacedName, &repository)
+				return repository.Status.AWS.Status
+			}, timeout).Should(Equal(cloudformation.CreateComplete))
+		})
+
+		By("displaying an AWS stack id in resource status", func() {
+			Eventually(func() string {
+				_ = client.Get(ctx, resourceNamespacedName, &repository)
+				return repository.Status.AWS.ID
+			}).ShouldNot(BeEmpty())
+		})
+
+		By("displaying a stack name prefixed with cluster name in resource status", func() {
+			Eventually(func() string {
+				_ = client.Get(ctx, resourceNamespacedName, &repository)
+				return repository.Status.AWS.Name
+			}).Should(ContainSubstring("xxx-ecr-test-test-image"))
+		})
+
+		By("ensuring a finalizer is present on resource to prevent deletion", func() {
+			Eventually(func() []string {
+				_ = client.Get(ctx, resourceNamespacedName, &repository)
+				return repository.Finalizers
+			}).Should(ContainElement(cloudformation.Finalizer))
+		})
+
+		By("ensuring no DeletionTimestamp exists", func() {
+			Eventually(func() bool {
+				_ = client.Get(ctx, resourceNamespacedName, &repository)
+				return repository.ObjectMeta.DeletionTimestamp == nil
+			}).Should(BeTrue())
+		})
+
+		By("creating a secret with repository details", func() {
+			Eventually(func() map[string][]byte {
+				_ = client.Get(ctx, secretNamespacedName, &secret)
+				return secret.Data
+			}).Should(And(
```

Comment:
`HasKey("ImageRegistryUsername"), HasKey("ImageRegistryPassword")` ?

Reply:
`ImageRegistry*` variables are provisioned by the PrincipalController not the ImageRepositoryController .... (credentials are per *Role* not per Repository)